src/cmd/ksh93/tests/read.exp

```diff
@@ -30,6 +30,18 @@ expect -ex "\r\ngimme input > "
 expect -ex "\r\ngot ''\r\n"
 expect_prompt
 
+#-p prompt Write prompt on each line before reading.
+send "read -p 'custom_prompt> ' foo\r"
+expect -ex "custom_prompt> "
```

Comment:
Writing `expect` based tests is hard. Here you need to consider that the input buffer will contain `custom_prompt>` twice. This will match the first instance which is from the previous `send`, not the actual prompt from executing the `read`. What I often do is deliberately cause the test to fail. For example, by introducing a typo in the expect pattern. Then looking at the log file to see what was actually in the buffer and tailoring the pattern to match what we're really interested in. Which in this case is the actual prompt.

What you really want here is `expect -re "\r\ncustom_prompt> $"`.